larrecodnn/ImagePatternAlgs/Tensorflow/WaveformRecogTools/IWaveformRecog.h

```diff
@@ -13,6 +14,182 @@ namespace wavrec_tool
 
         // Calculate multi-class probabilities for waveform
         virtual std::vector< std::vector<float> > predictWaveformType( const std::vector< std::vector<float> >& ) const = 0;
+
+        // ---------------------------------------------------------------------
+        // Return a vector of booleans of the same size as the input  waveform.
+	// The value of each element of the vector represents whether the 
+	// corresponding time bin of the waveform is in an ROI or not.
+        // ---------------------------------------------------------------------
+        std::vector<bool> findROI(const std::vector<float>& adcin) const
+        {
+          std::vector<bool>bvec(fWaveformSize,false);
+          if(adcin.size()!=fWaveformSize){
+            return bvec;
+          }
+
+          std::vector<std::vector<float>> predv = scanWaveform(adcin);
+
+          // .. set to true all bins in the output vector that are in windows identified as signals
+          int j1;
+	  for (unsigned int i=0; i<fNumStrides; i++){
+            j1 = i*fStrideLength;
+            if(predv[i][0]>fCnnPredCut){
+              std::fill_n(bvec.begin()+j1,fWindowSize,true);
+            }
+          }
+          // .. last window is a special case
+          if(predv[fNumStrides][0]>fCnnPredCut){
+            j1 = fNumStrides*fStrideLength;
+            std::fill_n(bvec.begin()+j1,fLastWindowSize,true);
+          }
+          return bvec;
+        }
+
+        // -------------------------------------------------------------
+        // Return a vector of floats of the same size as the input 
+	// waveform. The value in each bin represents the probability
+	// whether that bin is in an ROI or not
+        // -------------------------------------------------------------
+	std::vector<float> predROI(const std::vector<float>& adcin) const
+        {
+          std::vector<float>fvec(fWaveformSize,0.);
+          if(adcin.size()!=fWaveformSize){
+            return fvec;
+          }
+
+          std::vector<std::vector<float>> predv = scanWaveform(adcin);
+
+          // .. set value in each bin of output vector to the prediction for the window it is in
+          int j1;
+	  for (unsigned int i=0; i<fNumStrides; i++){
+            j1 = i*fStrideLength;
+            std::fill_n(fvec.begin()+j1,fWindowSize,predv[i][0]);
+          }
+          // .. last window is a special case
+          j1 = fNumStrides*fStrideLength;
+          std::fill_n(fvec.begin()+j1,fLastWindowSize,predv[fNumStrides][0]);
+          return fvec;
+        }
+
+    protected:
```

Comment:
It would be preferable if this could be `private`.  Is there a reason it cannot be?

Reply:
not sure you can do this since both protected methods findFile and setupWaveRecRoiParams are used by derived classes WaveformRecogTf and WaveformRecogTrtis

Reply:
Fair enough.

Reply:
I just noticed I forgot to remove the following line from WaveformRecogTf_tool.cc:
#include <sys/stat.h>
This is no longer needed.

Reply:
We'll get it next time, Mike.  No worries. :)